lib/api.js

```diff
@@ -211,10 +216,15 @@ class Api extends Emittery {
 
 				const options = {
 					...apiOptions,
+					filter,
 					providerStates,
 					recordNewSnapshots: !isCi,
 					// If we're looking for matches, run every single test process in exclusive-only mode
-					runOnlyExclusive: apiOptions.match.length > 0 || runtimeOptions.runOnlyExclusive === true
+					runOnlyExclusive: apiOptions.match.length > 0 || runtimeOptions.runOnlyExclusive === true,
+					runOnlySelectedLines: filter.some(({pattern, lineNumbers}) =>
```

Comment:
Rather than a boolean, what if we pass an array of line numbers? We're already checking for the filter items that match the test file anyway. And then we don't have to pass `filter` above.

Reply:
Works for me!

Reply:
Done. I still compute `runOnlySelectedLines` (= `lineNumbers.length > 0`) in `runner.js` as that is used in so many places.